common/ferrostar/src/navigation_controller/recording.rs

```diff
@@ -3,6 +3,10 @@ use crate::navigation_controller::models::{
     NavigationControllerConfig, NavigationRecordingEvent, NavigationRecordingEventData, TripState,
 };
 use chrono::Utc;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Object))]
 pub struct NavigationRecording {
```

Comment:
Question: does this need to be a public struct + UniFFI object? I'm not really sure if the callers need to care about this implementation detail. I guess they would just want to get the JSON data.